ros/processors/report_processor.py

```diff
@@ -0,0 +1,110 @@
+import json
+import requests
+import logging
+import tarfile
+from io import BytesIO
+from http import HTTPStatus
+from confluent_kafka import Consumer
+from ros.lib.host_inventory import fetch_host_from_inventory
+from ros.app import app, db
+from ros.config import INSIGHTS_KAFKA_ADDRESS
+from ros.models import PerformanceProfile
+
+LOG = logging.getLogger(__name__)
+running = True
+
+
+consumer = Consumer({
+    'bootstrap.servers': INSIGHTS_KAFKA_ADDRESS,
+    'group.id': 'ros-consumers',
+    "enable.auto.commit": False
+})
+
+
+class ReportProcessor:
+    def init_consumer(self):
+        consumer.subscribe(['platform.upload.resource-optimization'])
+        while running:
+            msg = consumer.poll(1.0)
+            if msg is None:
+                continue
+            if msg.error():
+                print(msg.error())
+                continue
+            try:
+                self.msg = json.loads(msg.value().decode("utf-8"))
+                self.handle_message()
+            except json.decoder.JSONDecodeError:
+                print(f"Unable to decode kafka message - {msg.value()}")
+            except Exception as err:
+                print(f"An error occurred during message processing - {repr(err)}")
+            finally:
+                consumer.commit()
+
+        consumer.close()
+
+    def handle_message(self):
+        self.report_url = self.msg.get('url', None)
+        metadata = self.msg.get('metadata', None)
+        if not metadata or not metadata['insights_id']:
+            return None
+        insights_id = metadata['insights_id']
+        rh_identity = self.msg.get('b64_identity', None)
+        host = fetch_host_from_inventory(insights_id, rh_identity)
+        if not host['results']:
+            print("No record found. Make sure system is registered in insights")
+            return None
+        if not self.report_url:
+            print("kafka message missing report url")
+        host_id = host['results'][0]['id']
+        report_tar = self._download_report()
+        performance_record = self._extract_performance_record(report_tar)
+
+        with app.app_context():
+            if performance_record:
+                performance_score = self._calculate_performance_score(performance_record)
+                profile = PerformanceProfile.query.filter_by(inventory_id=host_id).first()
+                if profile:
+                    profile.performance_record = performance_record
+                    profile.performance_score = performance_score
+                else:
+                    record = PerformanceProfile(inventory_id=host_id,
+                                                performance_record=performance_record,
+                                                performance_score=performance_score)
+                    db.session.add(record)
+
+                db.session.commit()
```

Comment:
Should we catch the sqlalchemy errors, in case of any db exceptions?

Reply:
If the only aim is to handle the exception and printing the log then again main exception block [here](https://github.com/RedHatInsights/ros-backend/pull/5/files#diff-3e77a430e2ccfd114f1e233d804c9077c12c508197b9ad61ab3038781f799b7aR39) in the parent method will handle it. 
Let me know if you are thinking of any counter step to perform in case of sqlalchemy exception then we can add the separate exception handling for it.